support-frontend/assets/helpers/redux/checkout/personalDetails/contributionsSideEffects.ts

```diff
@@ -0,0 +1,55 @@
+import type { ContributionsStartListening } from 'helpers/redux/contributionsStore';
+import * as storage from 'helpers/storage/storage';
+import { enableOrDisableForm } from 'pages/contributions-landing/checkoutFormIsSubmittableActions';
+import {
+	setEmail,
+	setFirstName,
+	setLastName,
+	setUserTypeFromIdentityResponse,
+} from './actions';
+
+export function addPersonalDetailsSideEffects(
+	startListening: ContributionsStartListening,
+): void {
+	startListening({
+		actionCreator: setFirstName,
+		effect(_, listenerApi) {
+			enableOrDisableForm()(
+				listenerApi.dispatch,
+				listenerApi.getState.bind(listenerApi),
+			);
+		},
+	});
+
+	startListening({
+		actionCreator: setLastName,
```

Comment:
:hand: It looks like you're doing the exact same side effect in multiple cases (either the `setFirstName` or `setLastName` action being dispatched).

You can use RTK's "matching" utilities to run the effect in both cases:

```js
const isNameUpdateAction = isAnyOf(setFirstName, setLastName);

startListening({
  matcher: isNameUpdateAction,
  effect: (action, listenerApi) => {
    dispatch(enableOrDisableForm())
  }
})
```

See https://redux-toolkit.js.org/api/matching-utilities#isanyof and https://redux-toolkit.js.org/api/createListenerMiddleware#startlistening